src/bitcoin-cli.cpp

```diff
@@ -191,7 +192,9 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
 
 UniValue CallRPC(const std::string& strMethod, const UniValue& params)
 {
-    std::string host = GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
+    std::string host;
+    int dummy_port;
+    SplitHostPort(GetArg("-rpcconnect", DEFAULT_RPCCONNECT), dummy_port, host);
```

Comment:
If you're going to do this anyway, why not allow overriding the port? (e.g. pass port instead of dummy_port)